examples/jsm/nodes/shadernode/ShaderNodeBaseElements.js

```diff
@@ -0,0 +1,257 @@
+import UniformNode from '../core/UniformNode.js';
+import VarNode from '../core/VarNode.js';
+import VaryNode from '../core/VaryNode.js';
+
+// accessors
+import BufferNode from '../accessors/BufferNode.js';
+import CameraNode from '../accessors/CameraNode.js';
+import MaterialNode from '../accessors/MaterialNode.js';
+import MaterialReferenceNode from '../accessors/MaterialReferenceNode.js';
+import ModelViewProjectionNode from '../accessors/ModelViewProjectionNode.js';
+import NormalNode from '../accessors/NormalNode.js';
+import Object3DNode from '../accessors/Object3DNode.js';
+import PointUVNode from '../accessors/PointUVNode.js';
+import PositionNode from '../accessors/PositionNode.js';
+import ReferenceNode from '../accessors/ReferenceNode.js';
+import TextureNode from '../accessors/TextureNode.js';
+import UVNode from '../accessors/UVNode.js';
+
+// math
+import MathNode from '../math/MathNode.js';
+import OperatorNode from '../math/OperatorNode.js';
+import CondNode from '../math/CondNode.js';
+
+// lights
+import ReflectedLightNode from '../lights/ReflectedLightNode.js';
+
+// utils
+import ArrayElementNode from '../utils/ArrayElementNode.js';
+import ConvertNode from '../utils/ConvertNode.js';
+
+// shader node utils
+import { ShaderNode, nodeObject, nodeObjects, nodeArray, nodeProxy, nodeImmutable, ConvertType, getConstNodeType, cacheMaps } from './ShaderNodeUtils.js';
+
+// shader node utils
+
+export { ShaderNode, nodeObject, nodeObjects, nodeArray, nodeProxy, nodeImmutable };
+
+export const color = new ConvertType( 'color' );
+
+export const float = new ConvertType( 'float', cacheMaps.float );
+export const int = new ConvertType( 'int', cacheMaps.int );
+export const uint = new ConvertType( 'uint', cacheMaps.uint );
+export const bool = new ConvertType( 'bool', cacheMaps.bool );
+
+export const vec2 = new ConvertType( 'vec2' );
+export const ivec2 = new ConvertType( 'ivec2' );
+export const uvec2 = new ConvertType( 'uvec2' );
+export const bvec2 = new ConvertType( 'bvec2' );
+
+export const vec3 = new ConvertType( 'vec3' );
+export const ivec3 = new ConvertType( 'ivec3' );
+export const uvec3 = new ConvertType( 'uvec3' );
+export const bvec3 = new ConvertType( 'bvec3' );
+
+export const vec4 = new ConvertType( 'vec4' );
+export const ivec4 = new ConvertType( 'ivec4' );
+export const uvec4 = new ConvertType( 'uvec4' );
+export const bvec4 = new ConvertType( 'bvec4' );
+
+export const mat3 = new ConvertType( 'mat3' );
+export const imat3 = new ConvertType( 'imat3' );
+export const umat3 = new ConvertType( 'umat3' );
+export const bmat3 = new ConvertType( 'bmat3' );
+
+export const mat4 = new ConvertType( 'mat4' );
+export const imat4 = new ConvertType( 'imat4' );
+export const umat4 = new ConvertType( 'umat4' );
+export const bmat4 = new ConvertType( 'bmat4' );
+
+// core
+
+// @TODO: ArrayUniformNode
+
+export const toFunction = ( node ) => {
+
+	node = nodeObject( new FunctionNode( node.code || node ) );
```

Comment:
Can you revert this function to `func`? The input parameter must be ever string. Like https://github.com/mrdoob/three.js/pull/23905/commits/bf2db826b5bcd88d6c196729162f752c46228bc6

Reply:
Will rename it. It supports both string and CodeNode as input.

Reply:
I see, but I think unnecessary, and the function signature `func( node )` is a little confuse.

Reply:
Is it better now?

Reply:
> Is it better now?

I think that just `func( code )`  is better.

Reply:
OK.